### DISCUSSION TITLE
Should MetricProducer be in the spec?

### DISCUSSION THREAD

**Discussion by anuraaga** [open-telemetry/opentelemetry-java#4240] [Q&A]
I was trying to match up the current metrics SDK implementation vs the spec and one interface I was surprised not to find there is `MetricProducer`.

https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/export/MetricProducer.java#L20
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#metricexporter

Is this something we would expect to be there? /cc @jsuereth

[the remark this answers is not in the document]
**Reply by reyang** [reply]
My guess - `MetricProducer` is a Java specific thing, which could be useful for bridging other systems such as Micrometer.

At least for now I don't see that the SDK spec will take it.

[the remark this answers is not in the document]
**Reply by anuraaga** [reply]
@reyang Micrometer shim doesn't use `MetricProducer`, it uses the API directly. OpenCensus shim does use it, but in that case I would expect it to be in the spec. I'm not aware of anything java-specific about this type - if it's just an ordering issue, that opencensus hasn't been spec'd yet, I can see if there's a way to move that into an internal type.

[the remark this answers is not in the document]
**Reply by reyang** [reply]
@anuraaga what you said makes sense to me 👍

[the remark this answers is not in the document]
**Reply by reyang** [reply]
FYI - there was something similar in OpenTelemetry .NET https://github.com/open-telemetry/opentelemetry-dotnet/pull/2683.
